index.d.ts

```diff
@@ -0,0 +1,129 @@
+/** 
+ * Typings for twitter-lite 
+ * 
+ * @version 0.9.4
+ * @author Floris de Bijl <@fdebijl>
+*/
+
+/// <reference types="node" />
+
+import { EventEmitter } from 'events';
+import { Header } from 'oauth-1.0a';
+
+declare namespace TwitterLite {
+  type TwitterOptions = {
+    /** "api" is the default (change for other subdomains) */
+    subdomain: string;
+    /** version "1.1" is the default (change for other subdomains) */
+    version: string;
+    /** consumer key from Twitter. */
+    consumer_key: string;
+    /** consumer secret from Twitter */
+    consumer_secret: string;
+    /** access token key from your User (oauth_token) */
+    access_token_key: string;
+    /** access token secret from your User (oauth_token_secret) */
+    access_token_secret: string;
+  };
+
+  type AccessTokenOptions = {
+    key: string;
+    secret: string;
+    /** If using the OAuth web-flow, set this parameter to the value of the oauth_verifier returned in the callback URL. If you are using out-of-band OAuth, set this value to the pin-code. */
+    verifier: string | number;
+  }
+
+  type BearerResponse = {
+    token_type: 'bearer';
+    access_token: string;
+  }
+
+  type RequestTokenResponse = {
+    oauth_token: string;
```

Comment:
oauth_token can be extracted to be its own type
and then used here and below.

Reply:
Could you give an example? In my understanding that would just make it an alias to `string`.

Reply:
@Fdebijl I think that's what he means, and I agree with the the reasoning. Any function parameter that repeats should probably be pulled out into it's own type for extensibility purposes (as another example, the `resource` parameter in `GET`, `POST`, and `PUT`). I'm not super familiar with TypeScript patterns, though, so take what I say with a grain of salt.

Reply:
An example would be
``` 
type oauth_token = string;
```
and then simply using it as 
```
type RequestTokenResponse = {
    oauth_token: oauth_token;
}
@dylanirlbeck Yea that's pretty much the purpose, centralizing it and benefiting from anything that comes naturally from centralizing stuff.